include/zephyr/bluetooth/services/ots.h

```diff
@@ -1029,6 +1056,23 @@ int bt_ots_client_write_object_data(struct bt_ots_client *otc_inst, struct bt_co
 				    const void *buf, size_t len, off_t offset,
 				    enum bt_ots_oacp_write_op_mode mode);
 
+/** @brief Get the checksum of the current selected object.
+ *
+ *  This will trigger an OACP calculate checksum operation for the current object
+ *  with a specified offset and length.
+ *
+ *  The checksum goes to OACP IND and obj_data_checksum() callback.
+ *
+ *  @param otc_inst     Pointer to the OTC instance.
+ *  @param conn         Pointer to the connection object.
+ *  @param offset       Offset to calculate, usually 0.
+ *  @param len          Len of data.
```

Comment:
Is there any way that the client can request the checksum without knowing the length of the object?

Reply:
No, I think it is a must.
Quote from OTS SPEC 

> A Length parameter is required with the Calculate Checksum Op Code. The object contents included in the calculation shall consist of Length number of octets, starting with the octet identified by the value of the Offset parameter.

Length can be obtained by Object Size characteristic.

Reply:
I see.

Given that we can do a checksum on a subset of the object, I guess `len` just needs to be `<= object_size`.